ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdStatistics.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015 Avanza Bank AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.avanza.ymer;
+
+import java.util.Set;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.openspaces.core.util.ConcurrentHashSet;
+
+class PersistedInstanceIdStatistics implements PersistedInstanceIdStatisticsMBean {
+
+	private final Set<Integer> readyForNumberOfPartitions = new ConcurrentHashSet<>();
+	private LongAdder calculationProgress;
+	private Long totalToCalculate;
+
+	public void resetStatisticsForJobExecution(Set<Integer> calculatingForPartitions) {
+		calculationProgress = new LongAdder();
+		totalToCalculate = 0L;
+		readyForNumberOfPartitions.removeIf((numberOfPartitions -> !calculatingForPartitions.contains(numberOfPartitions)));
```

Comment:
If `readyForNumberOfPartitions` is `[3, 4]` and `calculatingForPartitions` is `[4, 5]`, then this will result in `readyForNumberOfPartitions` becoming `[4]`, is this intended? Or should this be written as `readyForNumberOfPartitions.removeAll(calculatingForPartitions)`?

Reply:
Yes, this was intended. In a state where `[3, 4]` is already calculated and a new calculation is done for `[4, 5]`, it will still be ready for loading `[4]`. The index for `[3]` will be dropped when the calculation is started and will not work anymore. `[5]` will not be ready until the calculation is completed. However, if the current amount of partitions is `[4]`, it should be no problem restarting the instance, and continuing the calculation for `[5]` partitions where it ended.

`readyForNumberOfPartitions` is intended to show the number of instances the collection at this moment is ready for being restarted for.

Reply:
ok!